ClusteredMapView.js

```diff
@@ -43,15 +43,16 @@ export default class ClusteredMapView extends PureComponent {
     this.clusterize(this.props.data)
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (this.props.data !== nextProps.data)
-      this.clusterize(nextProps.data)
+  componentDidUpdate(prevProps) {
+    if (this.props.data !== prevProps.data) this.clusterize(this.props.data);
```

Comment:
I don't think this is going to be ideal because it will update state again after the component just rendered.

Reply:
Should it be `getDerivedStateFromProps` instead of `componentDidUpdate`? https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#new-lifecycle-getderivedstatefromprops

Reply:
Thank you for raising this, we surely have to work on redesigning the usage of React lifecycle to work well with React 16.x.x and onward. 

Yes it should probably be `getDerivedStateFromProps`.

Reply:
Please follow this issue #112 . We've made some changes and prepared a pre-release.